### PR TITLE
[wip] Load messages on demand

### PR DESCRIPTION
**Pull request by karissa**
this is not 100% working.

### CODE REVIEW

File: src/renderer/components/ChatView.js [hunk 1]
```diff
@@ -6,6 +6,8 @@ const SetupMessageDialog = require('./dialogs/SetupMessage')
 const Composer = require('./Composer')
 const { Overlay } = require('@blueprintjs/core')
 
+const { TrackVisibility } = require('react-on-screen')
```
Comment (ralphtheninja) [inline src/renderer/components/ChatView.js]: cool!

**Comment by karissa** [comment]
this is hard

**Comment by Jikstra** [comment]
Any ideas why the list is behaving weirdly (jumps around/renders again and again)?

**Comment by karissa** [comment]
@Jikstra something to do with estimating properly (and updating) the vertical height of each message..

I'm suspecting this is why signal renders ~20-30 messages at a time, then loads the next 20-30 once the person has scrolled all the way to the top, rather than estimating the scroll bar height and simulating virtual scroll. It's a bit easier to reason about and implement. We wouldn't need this 'infinitescroll' component then I think..

**Comment by karissa** [comment]
@Jikstra regarding height of each message, see: https://github.com/deltachat/deltachat-desktop/pull/261/commits/ab8660c5cc65a98ffc0f5591e343d78fa7cfafa2

**Comment by Jikstra** [comment]
> I'm suspecting this is why signal renders ~20-30 messages at a time, then loads the next 20-30 once the person has scrolled all the way to the top, rather than estimating the scroll bar height and simulating virtual scroll. It's a bit easier to reason about and implement. We wouldn't need this 'infinitescroll' component then I think..

But i think signal keeps all the messages rendered and just applies more elements to the ul. I don't know how well this scales if we want to rerender on every prepend.

**Comment by karissa** [comment]
closing in favor of #299

**Pull request closed without merging**